modules/OCPP/OCPP.cpp

```diff
@@ -57,6 +62,72 @@ void OCPP::init() {
             return false;
         }
     });
+    this->charge_point->register_upload_diagnostics_callback([this](std::string location) {
+        // create temporary file
+        std::string date_time = "DATETIME"; // FIXME
+        std::string file_name = "diagnostics-" + date_time + "-%%%%-%%%%-%%%%-%%%%";
+
+        auto diagnostics_file_name = boost::filesystem::unique_path(boost::filesystem::path(file_name));
+        auto diagnostics_file_path = boost::filesystem::temp_directory_path() / diagnostics_file_name;
+
+        auto diagnostics = json::object({{"diagnostics", json::object({{"key", "value"}})}});
+        std::ofstream diagnostics_file(diagnostics_file_path.c_str());
+        diagnostics_file << diagnostics.dump();
+        this->upload_diagnostics_thread = std::thread([this, location, diagnostics_file_name, diagnostics_file_path]() {
+            auto diagnostics_uploader = boost::filesystem::path("bin/diagnostics_uploader.sh");
+
+            boost::process::ipstream stream;
+            std::vector<std::string> args = {location, diagnostics_file_name.string(), diagnostics_file_path.string()};
+            boost::process::child cmd(diagnostics_uploader, boost::process::args(args),
+                                      boost::process::std_out > stream);
+            std::string temp;
+            while (std::getline(stream, temp)) {
```

Comment:
that's a nice solution. Probably I should have also used this boost process functionality in the manager.cpp